### DISCUSSION TITLE
Petsc ASPIN solver

### DISCUSSION THREAD

**Discussion by andrembcosta** [idaholab/moose#22160] [Q&A General] [answered]
Hi there,

I came across a paper where an Additive Schwartz nonlinear preconditioner is used for phase-field fracture problems and it seems to work quite well. Apparently, this type of nonlinear preconditioner already exists in petsc. (https://petsc.org/release/docs/manualpages/SNES/SNESASPIN.html)

Has someone used this specific solver within moose? I am interested in trying it out, but I couldn't find an example with the syntax on how to define the blocks and how to modify the residual computation (to account for the nonlinear preconditioner).

I just observed that the author of the paper asked this question in early 2019 (https://groups.google.com/g/moose-users/c/w19VwqHtspU/m/cvzpirFABQAJ).

paper: https://www.researchgate.net/publication/359439046_Nonlinear_Field-split_Preconditioners_for_Solving_Monolithic_Phase-field_Models_of_Brittle_Fracture

**Comment by hugary1995** [comment, marked as answer]
@andrembcosta could you also post the link to that paper here? I'm also interested in using this solver.

**Reply by andrembcosta** [reply]
https://www.researchgate.net/publication/359439046_Nonlinear_Field-split_Preconditioners_for_Solving_Monolithic_Phase-field_Models_of_Brittle_Fracture

**Reply by lindsayad** [reply]
I'll mention @fdkong here in case he wants to chime in. Otherwise hopefully I can weigh-in with a more informed answer in not too long

**Reply by lindsayad** [reply]
so I think that supporting ASPIN is definitely possible. Doing it in the way that is done in the paper you linked will probably be much easier after https://github.com/idaholab/moose/pull/22226 is complete as that will provide a very straightforward way to define subproblems based on variables. Note that I have created #22239 to keep track of progress made on supporting ASPIN
